### PR TITLE
[i2c, rtl] Update i2c.prj.hjson for D2 completion

### PR DESCRIPTION
**Pull request by igk8**
Signed-off-by: Igor Kouznetsov <igor.kouznetsov@wdc.com>

### CODE REVIEW

**Review by tjaychen — APPROVED** [review]

**Comment by martin-lueker** [comment]
This is great news @igk8!

All: Was there a review meeting for this?  Do we need one?   (I know that D1 is pretty relaxed, but I hear some requests for D2/V2 reviews)

File: hw/ip/i2c/data/i2c.prj.hjson [hunk 1]
```diff
@@ -12,10 +12,10 @@
       {
         version:            "0.5",
         life_stage:         "L1",
-        design_stage:       "D1",
+        design_stage:       "D2",
         verification_stage: "V1",
         dif_stage:          "S0",
-        commit_id:          "ff9c06167bc34d921974839017dec54f0fb8347a",
+        commit_id:          "a44bdf910e4941a40a2bccfbecb525ac0b68f727",
```
Comment (msfschaffner) [inline hw/ip/i2c/data/i2c.prj.hjson:18]: The commit ID is not needed as long as the design is still under development.
It only needs to be added when we roll back design stages (e.g., go back from D2 to D1) to keep the latest snapshot of the previous design.
Hence, this line can be removed.

**Review by msfschaffner — APPROVED** [review]
Since the checklist has already passed GitHub review, this should be ok to merge (modulo my comment below).
For D3 a formal design review needs to happen with representatives from all stakeholders in the room.

File: hw/ip/i2c/data/i2c.prj.hjson (hunk 1 above)
Comment (igk8) [reply hw/ip/i2c/data/i2c.prj.hjson]: This line is removed.

**Pull request merged**